reactor-test/src/main/java/reactor/test/util/TestLogger.java

```diff
@@ -166,17 +216,37 @@ public boolean isErrorEnabled() {
 
 	@Override
 	public synchronized void error(String msg) {
-		this.err.format("[ERROR] (%s) %s\n", Thread.currentThread().getName(), msg);
+		if(logCurrentThreadName){
+			this.err.format("[ERROR] (%s) %s\n", Thread.currentThread().getName(), msg);
+		} else {
+			this.err.format("[ERROR] %s\n", msg);
+		}
 	}
 
 	@Override
 	public synchronized void error(String format, Object... arguments) {
-		this.err.format("[ERROR] (%s) %s\n", Thread.currentThread().getName(), format(format, arguments));
+		if(logCurrentThreadName){
+			this.err.format("[ERROR] (%s) %s\n", Thread.currentThread().getName(), format(format, arguments));
+		} else {
+			this.err.format("[ERROR] %s\n", format(format, arguments));
+		}
 	}
 
 	@Override
 	public synchronized void error(String msg, Throwable t) {
-		this.err.format("[ERROR] (%s) %s - %s\n", Thread.currentThread().getName(), msg, t);
+		if(logCurrentThreadName){
+			this.err.format("[ERROR] (%s) %s - %s\n", Thread.currentThread().getName(), msg, t);
+		} else {
+			this.err.format("[ERROR] %s - %s\n", msg, t);
+		}
 		t.printStackTrace(this.err);
 	}
+
+	public boolean isLogCurrentThreadName() {
+		return logCurrentThreadName;
+	}
+
+	public void setLogCurrentThreadName(boolean logCurrentThreadName) {
```

Comment:
I don't think there is much need to make it mutable. could you replace the setter with a second constructor taking the boolean directly, and make the field `final`?